clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy

```diff
@@ -177,14 +184,74 @@ class KubernetesClientApiAdapter {
     }
   }
 
+  private Map[] determineJsonPatch(Object current, Object desired) {
+    JsonNode desiredNode = mapper.convertValue(desired, JsonNode.class);
+    JsonNode currentNode = mapper.convertValue(current, JsonNode.class);
+
+    return mapper.convertValue(JsonDiff.asJson(currentNode, desiredNode), Map[].class);
+  }
+
+  V1beta1StatefulSet resizeStatefulSet(String name, String namespace, int targetSize, boolean mustWait) {
+    exceptionWrapper("statefulSets.resize", "Resize Stateful Set $name", namespace) {
+      V1beta1StatefulSet current = getStatefulSet(name, namespace)
+      V1beta1StatefulSet desired = getStatefulSet(name, namespace)
+      desired.spec.replicas = targetSize
+
+      final Map[] jsonPatch = determineJsonPatch(current, desired);
+      V1beta1StatefulSet statefulSet = apiInstance.patchNamespacedStatefulSet(name, namespace, jsonPatch, null)
+
+      if (mustWait) {
+        def labels = statefulSet.metadata.labels
+        V1PodList pods = getPods(namespace, labels)
+        int podSize = pods.items.size()
+        int terminationGracePeriodSeconds = statefulSet.spec.template.spec.terminationGracePeriodSeconds ?: TERMINATION_GRACE_PERIOD_SECONDS
+
+        if (podSize != targetSize) {
+          waitForNumberOfPodUpdated(name, namespace, labels, podSize, targetSize, terminationGracePeriodSeconds)
```

Comment:
This still shouldn't be happening here. We need orca to poll for this. If a clouddriver node goes down during this operation we lose all state, and given that this operation blocks we are creating a large window for that to happen.

Reply:
We need to remove all running pods before we update the statefulset(note: this is for update strategy not resize).  I don't think orca can handle because these are consecutively steps.  It is not just monitor the pods status can do.  If the pod does not shudown, and no new pods will be update.

Reply:
I don't see why orca can't do that, usually it orchestrates consecutive steps for more complicated operations.

Reply:
unless that you want to resize first before update.  Then, orca needs to know if this is a newly create or update.